Detectors/FIT/FT0/raw/include/FT0Raw/DataBlockRaw.h

```diff
@@ -0,0 +1,429 @@
+        payloadPerElem = T::PayloadSize;
+      payloadPerElem -= std::get<kNBYTES>((GetByteLookupTable())[iStep]);
+    }
+    //Last step checking
+    std::get<kNSTEPS>(readingScheme[countWord]) = getNsteps();
+    if (payloadPerElem > 0) {
+      std::get<kISPARTED>(readingScheme[countWord]) = true;
+      std::get<kNELEMENTS>(readingScheme[countWord]) = std::get<kELEMENTINDEX>((GetByteLookupTable())[getNsteps() - 1]);
+    } else {
+      std::get<kISPARTED>(readingScheme[countWord]) = false;
+      std::get<kNELEMENTS>(readingScheme[countWord]) = std::get<kELEMENTINDEX>((GetByteLookupTable())[getNsteps() - 1]) + 1;
+    }
+    return readingScheme;
+  }
+  static constexpr std::array<std::tuple<unsigned int, unsigned int, bool>, MaxNwords> sReadingLookupTable = GetReadingLookupTable();
+  //
+  //Printing LookupTables
+  static void printLUT()
+  {
+    cout << endl
+         << "-------------------------------------------" << endl;
+    std::cout << "kNELEMENTS|kNSTEPS|kISPARTED" << std::endl;
+    for (int i = 0; i < MaxNwords; i++) {
+      std::cout << std::endl
+                << std::get<kNELEMENTS>(sReadingLookupTable[i]) << "|"
+                << std::get<kNSTEPS>(sReadingLookupTable[i]) << "|"
+                << std::get<kISPARTED>(sReadingLookupTable[i]) << endl;
+    }
+    cout << endl
+         << "-------------------------------------------" << endl;
+    std::cout << "kELEMENTINDEX|kWORDINDEX|kNBYTES|kSRCBYTEPOS|kDESTBYTEPOS" << std::endl;
+    for (int i = 0; i < getNsteps(); i++) {
+      cout << endl
+           << std::get<kELEMENTINDEX>(sByteLookupTable[i]) << "|"
+           << std::get<kWORDINDEX>(sByteLookupTable[i]) << "|"
+           << std::get<kNBYTES>(sByteLookupTable[i]) << "|"
+           << std::get<kSRCBYTEPOS>(sByteLookupTable[i]) << "|"
+           << std::get<kDESTBYTEPOS>(sByteLookupTable[i]) << endl;
+    }
+  }
+  void print()
+  {
+    assert(mNelements <= T::MaxNelements);
+    for (int i = 0; i < mNelements; i++) {
+      std::cout << "\nPringting element number: " << i << std::endl;
+      mData[i].print();
+    }
+  }
+  T mData[T::MaxNelements];
+  unsigned int mNelements; //number of deserialized elements;
+  unsigned int mNwords;    //number of deserialized GBT words;
+};
+
+//CRTP(static polymorphism) + Composition over multiple inheritance(Header + multiple data structures)
+template <class DataBlock, class Header, class... DataStructures>
+class DataBlockBase : public boost::mpl::inherit<DataBlockWrapper<Header>, DataBlockWrapper<DataStructures>...>::type
+{
+  typedef boost::mpl::vector<DataStructures...> DataBlockTypes;
+  typedef DataBlockBase<DataBlock, Header, DataStructures...> TemplateHeader;
+  typedef typename boost::mpl::inherit<DataBlockWrapper<Header>, DataBlockWrapper<DataStructures>...>::type DataBlockDerivedBase;
+
+ public:
+  DataBlockBase() = default;
+  DataBlockBase(const DataBlockBase& other) = default;
+
+  virtual ~DataBlockBase() = default;
```

Comment:
Hello @shahor02 !
I will fix this carefully, because of using CRTP(static polymorphism) there should be no virtual methods, I forget about destructors.